[DOC-3575] 80308 Delegate release notes (#2913)
harness/developer-hub: docs/first-gen/firstgen-release-notes/fg-delegate.md

<Kustomizedep />

## Latest: Harness version 80308, Harness Delegate version 80308

Harness FirstGen release 80308 includes the following changes for the Harness Delegate.

### New features and enhancements

- If you use the App Role authentication method in the Hashicorp Vault connector, you can choose to cache the vault token. The token is cached on the Harness Delegate for a time duration equal to the TTL of the vault token, with 1% leeway. 

  By default, caching is enabled for all existing connectors. To disable caching, go to the connector's YAML configuration and set the `enableCache` parameter to `false`. Harness UI support to enable and disable caching will be added in a subsequent release. (PL-39821)

### Early access features

This release does not include any new early access features.

### Fixed issues

- Fixed an issue observed in Canary deployments where the rollback stage could not identify and delete the canary workload in some clusters. (CDS-76240)

> - Fixed retries of the delegate task acquire call in Harness Manager. Harness Manager returned NPEs when retrying acquire calls because `taskDataV2` was not copied to `taskData` in the acquire call retry flow. Tasks timed out because the delegate was not able to acquire the data. The 'taskData' field in Harness Manager is now populated to fix the issue. (PL-40646)

Reviewer: ```suggestion
- Fixed the delegate task *acquire call* retry flow in Harness Manager. Harness Manager returned NPEs when retrying acquire calls because `taskDataV2` was not copied to `taskData` in the *acquire call* retry flow. Tasks timed out because the delegate couldn't acquire the data. To fix this issue, the 'taskData' field in Harness Manager is now populated. (PL-40646)
```

The first sentence is a little confusing due to the function name. I think this release note would be fine without the first sentence, but I also see that you are trying to follow a pattern.
